common/lib/opentelemetry/common/utilities.rb

```diff
@@ -101,6 +101,36 @@ def cleanse_url(url)
         url
       end
 
+      # Returns the first non nil environment variable requested,
+      # or the default value if provided.
+      #
+      # @param [String] env_vars The environment variable(s) to retrieve
+      # @param default The fallback value to return if the requested
+      #  env var(s) are not present
+      #
+      # @returns [String]
+      def config_opt(*env_vars, default: nil)
+        env_vars.each do |env_var|
+          val = ENV[env_var]
+          return val unless val.nil?
+        end
+        default
+      end
+
+      # Returns a true if the provided url is invalid
+      #
+      # @param [String] url the URL string to test validity
+      #
+      # @return [boolean]
+      def invalid_url?(url)
```

Comment:
I am inclined to lean towards positive method names instead of negative ones, i.e. `valid_url?` as opposed to `invalid_url?`. 

For convenience if we wanted to support the inverse I would appreciates a `valid_url?` combined with `def invalid_url?(url); !valid_url?(url); done`

Also it would really be good to capture something about the reason why the URL is invalid. That could give a clue to someone debugging this.

Reply:
No strong feelings towards positive vs negative methods here, so I can switch that.

Re: Capturing something about the reason.

Technically I think we could just let the error bubble up from creating the URI out of the url provided, and do away with this method entirely.  However I think the error message provided is a lot less human friendly in most cases, even if the stack trace sends you to the right place.

Here is a random example I used in the tests
```
pry(main)> URI('1113:1113')
URI::InvalidURIError: bad URI(is not URI?): "1113:1113"
from /opt/rubies/2.7.5/lib/ruby/2.7.0/uri/rfc3986_parser.rb:67:in `split'
```
vs

`"invalid url for OTLP::Exporter 1113:1113"`

Reply:
Follow up thought, if we wanted to provide more hints we can do that as a follow on. I'm just shuffling some code around right now :)